lib/broccoli/ember-app.js

```diff
@@ -250,7 +242,21 @@ EmberApp.prototype._initOptions = function(options) {
       vendor: vendorTree,
       public: publicTree
     }
-  }, DEFAULT_CONFIG);
+  };
+
+  var emberCLIBabelInstance = this.project.findAddonByName('ember-cli-babel');
+  if (emberCLIBabelInstance) {
+    // future versions of ember-cli-babel will be moving the location for its
+    // own configuration options out of `babel` and will be issuing a deprecation
+    // if used in the older way
+    //
+    // see: https://github.com/babel/ember-cli-babel/pull/105
+    var emberCLIBabelConfigKey = emberCLIBabelInstance.configKey || 'babel';
+    detectedOptions[emberCLIBabelConfigKey] = detectedOptions[emberCLIBabelConfigKey] || {};
```

Comment:
Can we not just go ahead and set both `ember-cli-babel` and `babel` inside of the `detectedOptions` hash? This feels obfuscating.